OptimizelySDK/Odp/OdpEventManager.cs

```
@@ -0,0 +1,448 @@
+        )
+        {
+            _odpConfig = odpConfig;
+            _odpEventApiManager = odpEventApiManager;
+            _logger = logger;
+            _queueSize = queueSize;
+            _batchSize = batchSize;
+            _flushInterval = flushInterval;
+
+            _queue = new ConcurrentQueue<OdpEvent>();
+            _flushIntervalCancellation = new CancellationTokenSource();
+            _flushQueueRegularly = new Task(RegularlyFlushQueue, _flushIntervalCancellation.Token);
+        }
+
+        /// <summary>
+        /// Update ODP configuration settings
+        /// </summary>
+        /// <param name="odpConfig">Configuration object containing new values</param>
+        public void UpdateSettings(OdpConfig odpConfig)
+        {
+            _odpConfig.Update(odpConfig.ApiKey, odpConfig.ApiHost, odpConfig.SegmentsToCheck);
+        }
+
+        /// <summary>
+        /// Start processing events in the queue
+        /// </summary>
+        public void Start()
+        {
+            if (!IsOdpConfigurationReady())
+            {
+                return;
+            }
+
+            CurrentState = ExecutionState.Running;
+
+            _flushQueueRegularly.Start();
+        }
+
+        /// <summary>
+        /// Drain the queue sending all remaining events in batches then stop processing
+        /// </summary>
+        public void Stop()
+        {
+            if (!IsOdpConfigurationReady())
+            {
+                return;
+            }
+
+            if (CurrentState == ExecutionState.Stopped)
+            {
+                return;
+            }
+
+            _logger.Log(LogLevel.DEBUG, "Stop requested.");
+
+            _flushIntervalCancellation.Cancel();
+            try
+            {
+                _flushQueueRegularly.ConfigureAwait(false).GetAwaiter().GetResult();
+            }
+            catch (OperationCanceledException)
+            {
+                // exception raised by design from .Cancel()
```

please log exception, don't swallow exceptions.